File: aoj/volume100/python3/A10033.py
import sys

stacks = []

no_stacks = int(input())
for _ in range(no_stacks):
    stacks.append([])

for operation in sys.stdin:
    operation = operation.strip()

    if operation.startswith('quit'):
        break

    elif operation.startswith('push'):
        _, index, block = operation.split()
        stacks[int(index)-1].append(block)

    elif operation.startswith('pop'):
        _, index = operation.split()
        print(stacks[int(index)-1].pop())

    elif operation.startswith('move'):
        _, src_index, dst_index = operation.split()
        stacks[int(dst_index)-1].append(stacks[int(src_index)-1].pop())

File: aoj/volume100/nodejs/A10033.js
function main(lines) {
  var noStacks = parseInt(lines[0], 10);
  var stacks = new Array(noStacks);
  for (var si = 0; si < noStacks; si++)
    stacks[si] = [];
  for (var i = 1;; i++) {
    var operation = lines[i];
    if (operation.match(/^quit/)) {
      break;
    } else if (operation.match(/^push/)) {
      var args = operation.split(/\s+/);
      var index = parseInt(args[1], 10) - 1;
      var block = args[2];
      stacks[index].push(block);
    } else if (operation.match(/^pop/)) {
      var args = operation.split(/\s+/);
      var index = parseInt(args[1], 10) - 1;
      var block = stacks[index].pop();
      console.log(block);
    } else if (operation.match(/^move/)) {
      var args = operation.split(/\s+/);
      var src = parseInt(args[1], 10) - 1;
      var dst = parseInt(args[2], 10) - 1;
      stacks[dst].push(stacks[src].pop())
    }
  }
}

var chunks = [];
process.stdin.on('data', function(chunk) {
  chunks.push(chunk)
});
process.stdin.on('end', function() {
  var lines = chunks.join('').split('\n');
  main(lines);
});
